Homework_10/src/Main.java

```diff
@@ -0,0 +1,48 @@
+public class Main {
+    public static void main(String[] args) {
+
+
+        User anna = new User("Anna");
+        User maria = new User("Maria");
+        User dmytro = new User("Dmytro");
+        User oleg = new User("Oleg");
+
+
+        anna.addContact(maria);
+        anna.addContact(dmytro);
+
+        maria.addContact(anna);
+
+        dmytro.addContact(maria);
+        dmytro.addContact(anna);
+
+
+        anna.sendMessage("Hello, how are you?", maria);
+        maria.sendMessage("Hi, what's new?", anna);
+        dmytro.sendMessage("Hi, how's your day going? ", anna);
+
+        showUserMessages(anna);
+        showUserMessages(maria);
+        showUserMessages(dmytro);
+
+        anna.getMessages().forEach(anna::readMessage);
+        dmytro.getMessages().forEach(dmytro::readMessage);
+        System.out.println();
+
+        showUserMessages(anna);
+        showUserMessages(maria);
+        showUserMessages(dmytro);
+
+    }
+
+    private static void showUserMessages(User user) {
```

Comment:
Works pretty well, although could be a method of a User itself, but anyway thumbs up for extracting this into the method so you don't need to do it many times in main()